ibu-imager/installation_configuration_files/scripts/installation-configuration.sh

```diff
@@ -225,5 +225,10 @@ until openssl x509 -in ${KUBELET_SERVING_CERTIFICATE} -checkend 30 &> /dev/null;
 done
 echo "${KUBELET_SERVING_CERTIFICATE} is valid."
 
+IBU_FILE=/var/ibu.yaml
+if oc apply -f ${IBU_FILE}; then
```

Comment:
@omertuc I need to replace this with a curl post. Do we already have the cacert and token for accessing api server in this script?

Reply:
You can grab all you need from the kubeconfig

Reply:
But it's fine keeping it `oc` since that's guaranteed to be available in this context and used by many other OCP scripts

Reply:
Why we should have ibu to be different than any other manifest?
Please add this logic to clusterconfig.go as it is part of it

Reply:
We already apply all manifests from cluster-configuration folder, there is no need to create new way for the same logic

Reply:
I'd like to preserve the CR status. Do you have an example somewhere that extracts things from kubeconfig?

Reply:
```
API_PATH=api/v1/pods
curl \
               $(oc config view --raw -ojson | jq '.clusters[0].cluster.server' -r)/$API_PATH \
    --cacert   <(oc config view --raw -ojson | jq '.clusters[0].cluster."certificate-authority-data"' -r | base64 -d) \
    --cert     <(oc config view --raw -ojson | jq '.users[0].user."client-certificate-data"' -r | base64 -d) \
    --key      <(oc config view --raw -ojson | jq '.users[0].user."client-key-data"' -r | base64 -d)
```

Reply:
This is just an example, we should be more robust with `.current-context` etc

Reply:
Wondering if we actually need status? Seems so many code, i assume i would like to understand why status is actually required as it will have really small amount of data and "verify after pivot" state can be done in code and i think there is already PR for it

Reply:
I mean this code doesn't worth this status or i miss something?
` conditions:
    - lastTransitionTime: "2023-11-15T14:52:50Z"
      message: In progress
      observedGeneration: 10
      reason: InProgress
      status: "False"
      type: Idle
    - lastTransitionTime: "2023-11-15T14:54:21Z"
      message: Prep completed
      observedGeneration: 10
      reason: Completed
      status: "True"
      type: PrepCompleted
    - lastTransitionTime: "2023-11-15T14:54:21Z"
      message: Prep completed
      observedGeneration: 10
      reason: Completed
      status: "False"
      type: PrepInProgress
    - lastTransitionTime: "2023-11-15T14:56:30Z"
      message: Upgrade completed
      observedGeneration: 11
      reason: Completed
      status: "True"
      type: UpgradeCompleted
    - lastTransitionTime: "2023-11-15T14:56:30Z"
      message: Upgrade completed
      observedGeneration: 11
      reason: Completed
      status: "False"
      type: UpgradeInProgress
    observedGeneration: 11
`

Reply:
This is only a lot of code because we're still using bash, this should be Go, then it's a matter of creating a simple kubernetes client which automatically loads the kubeconfig

Reply:
Yeah but this will require moving to go and this is actually i really hoped we will not need. 
Trying to understand why we need this status at all.
I got status from upgrade cr and this is exactly the one we want to preserve. Trying to understand why it is needed

Reply:
If we need this status we should move to golang cause this code is not nice

Reply:
Feel free to create a story for converting the whole thing to go.

Reply:
Go indeed. Lets create a jira ticket

Reply:
https://issues.redhat.com/browse/CNF-10403